lib/Myriad/Registry.pm

```diff
@@ -31,13 +31,19 @@ declare_exception UnknownClass => (
 
 use Myriad::API;
 
-has $rpc = {};
-has $service_by_name = {};
-has $batch = {};
-has $sink = {};
-has $stream = {};
-has $emitter = {};
-has $receiver = {};
+has $rpc ;
+has $service_by_name;
+has $batch;
+has $emitter;
+has $receiver;
+
+BUILD {
+ $rpc = {};
```

Comment:
Any particular reason why you're storing these as hashrefs? You could just

```perl
has %rpc;
```

and save yourself the trouble in the `BUILD` block